### DISCUSSION TITLE
Create record with association using prefix ids

### DISCUSSION THREAD

**Discussion by lostmarinero** [excid3/prefixed_ids#82] [Q&A] [answered]
New here, so apologies if this is covered somewhere else - did a search, dug into the code, and read old issues and did not find anything.

We are using this library, and have a User model and a Company model.

When we are trying to create a user with params like:
`{"company_id"=>"company_VmO1Gbv7BpXN", "first_name"=>"Test", "last_name"=>"User"`

We are getting an error:
`"Company must exist"`

But it does exist, if we do:
`Company.find("company_VmO1Gbv7BpXN")`
The company is found

Is it possible to create a record with an associated object using a prefix-id?

**Comment by excid3** [comment, marked as answer]
Correct, you'd have to intercept that and decode it to the primary key ID.

Your model can do that with something like this:

```ruby
def company_id=(value)
  value.start_with?(_prefix_id.prefix) ? super(_prefix_id.decode(value)) : super

  # Or with fallbacks enabled, should be able to just decode without the conditional:
  # super _prefix_id.decode(value, fallback: true).first
end
```

The more we override of ActiveRecord internals, the more issues arise so we've been trying to keep it minimal.

**Reply by lostmarinero** [reply]
Thank you for this info! And the quick response
